### PR TITLE
Replying to mentions

### PR DESCRIPTION
**Pull request by hfgd123**
As suggested by shini in https://features.sc-network.net/board/custombot/412 I added a reply-to-mention feature in the auto-react-module. Something went wrong with my Webstorm branches again, so the commits are made directly on the GitHub website, pasted from webstorm

### CODE REVIEW

**Comment by SCDerox** [comment]
Please add support for embeds using config elements

File: modules/auto-react/config.json [hunk 1]
```diff
@@ -29,6 +29,20 @@
...
     },
+    {
+      "field_name": "membersReply",
+      "default": {},
+      "humanname-de": "Erwähnungsantworten",
+      "humanname-en": "Mention-replys",
+      "type": "keyed",
+      "description-en": "Here you can add members and the reply on mentions of them",
+      "description-de": "Du kannst hier NutzerIDs und die Antwort auf Erwähnungen dieser eintragen",
+      "allowEmbed": true,
```
Comment (SCDerox) [inline modules/auto-react/config.json:40]: This is not supported, please use config-elements instead

**Review by SCDerox — CHANGES_REQUESTED** [review]

File: modules/auto-react/config.json (hunk 1 above)
Comment (hfgd123) [reply modules/auto-react/config.json:40]: ok. I have no idea how to use this 😅 I can create a working config with it, but how do I read from it?

Comment (SCDerox) [reply modules/auto-react/config.json:40]: Same way as usual files, but you get an array of objects with the configured values. 

Tipp: you can always use `console.log(value)` to see the content of a variable

File: modules/auto-react/events/messageCreate.js [hunk 2]
```diff
@@ -26,6 +29,27 @@ async function checkMembers(msg) {
...
 }
 
+/**
+ * Checks for member pings in a message and replys with the configured message
+ * @private
+ * @param msg
+ * @returns {Promise<void>}
+ */
+async function checkMembersReply(msg) {
+    const moduleConfig = msg.client.configurations['auto-react']['replys'];
```
Comment (SCDerox) [inline modules/auto-react/events/messageCreate.js:39]:
```suggestion
    const moduleConfig = msg.client.configurations['auto-react']['replies'];
```

This config file does not exist in your current project. Please commit.

**Review by SCDerox — CHANGES_REQUESTED** [review]

**Review by SCDerox — CHANGES_REQUESTED** [review]
Sorry, but this does not work at all - several issues such as comparing objects with IDs, wrong looping and others need to be addressed...
I will fix these issues (I spent way too much time trying to give you feedback, but at this point it's easier to do it myself completely, sorry).
You can find a ZIP of the working version attached.
[auto-react.zip](https://github.com/SCNetwork/CustomDCBot/files/8800610/auto-react.zip)

**Comment by hfgd123** [comment]
I shouldn't do anyting while I'm tired

**Comment by SCDerox** [comment]
No worries, I know this too well, if I am tired I can't do anything right ^^

**Pull request closed without merging**